artifactory/commands/mvn/mvn.go

```diff
@@ -238,3 +245,46 @@ func (mc *MvnCommand) conditionalUpload() error {
 	}
 	return err
 }
+
+// updateBuildInfoArtifactsWithTargetRepo updates existing build-info temp file with the target repository for each artifact
+func (mc *MvnCommand) updateBuildInfoArtifactsWithTargetRepo(vConfig *viper.Viper, buildInfoService *biservice.BuildInfoService) error {
+	buildName, err := mc.configuration.GetBuildName()
+	if err != nil {
+		return err
+	}
+	buildNumber, err := mc.configuration.GetBuildNumber()
+	if err != nil {
+		return err
+	}
+	mvnBuild, err := buildInfoService.GetOrCreateBuildWithProject(buildName, buildNumber, mc.configuration.GetProject())
+	if err != nil {
+		return errorutils.CheckError(err)
+	}
+	buildInfo, err := mvnBuild.ToBuildInfo()
+	if err != nil {
+		return errorutils.CheckError(err)
+	}
+
+	if vConfig.IsSet("deployer") {
+		snapshotRepository := vConfig.GetString("deployer.snapshotrepo")
+		releaseRepository := vConfig.GetString("deployer.releaserepo")
+		for moduleIndex := range buildInfo.Modules {
+			currModule := &buildInfo.Modules[moduleIndex]
+			for artifactIndex := range currModule.Artifacts {
+				updateArtifactRepo(&currModule.Artifacts[artifactIndex], snapshotRepository, releaseRepository)
+			}
+			for artifactIndex := range currModule.ExcludedArtifacts {
+				updateArtifactRepo(&currModule.ExcludedArtifacts[artifactIndex], snapshotRepository, releaseRepository)
+			}
+		}
+	}
+	return mvnBuild.SaveBuildInfo(buildInfo)
+}
+
+func updateArtifactRepo(artifact *entities.Artifact, snapshotRepo, releaseRepo string) {
+	if snapshotRepo != "" && strings.Contains(artifact.Path, "-SNAPSHOT") {
+		artifact.OriginalRepo = snapshotRepo
```

Comment:
Please update go.mod to dev branches and rename field to `OriginalDeploymentRepo`, thanks.